File: be/models/RepairItem.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const RepairItem = sequelize.define('RepairItem', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  repairTicketId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'repair_tickets',
      key: 'id'
    }
  },
  taskName: {
    type: DataTypes.STRING,
    allowNull: false
  },
  partName: {
    type: DataTypes.STRING,
    defaultValue: '---'
  },
  quantity: {
    type: DataTypes.INTEGER,
    allowNull: false,
    defaultValue: 1
  },
  partPrice: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    defaultValue: 0
  },
  laborPrice: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    defaultValue: 0
  },
  totalPrice: {
    type: DataTypes.DECIMAL(10, 2),
    allowNull: false,
    defaultValue: 0
  },
  isCompleted: {
    type: DataTypes.BOOLEAN,
    allowNull: false,
    defaultValue: false,
    comment: 'Whether this repair item has been completed'
  },
  completedAt: {
    type: DataTypes.DATE,
    allowNull: true,
    comment: 'Timestamp when this item was marked as completed'
  }
}, {
  tableName: 'repair_items',
  timestamps: true
});

module.exports = RepairItem;

File: be_python/models/repair_item.py
from sqlalchemy import Column, Integer, String, Numeric, Boolean, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from database.engine import Base


class RepairItem(Base):
    """
    Individual line item within a RepairTicket.
    Tracks part cost, labour cost, completion status, and timestamps.
    """

    __tablename__ = "repair_items"

    id = Column(Integer, primary_key=True, autoincrement=True)
    repairTicketId = Column(
        "repairTicketId", Integer, ForeignKey("repair_tickets.id"), nullable=False, index=True
    )
    taskName = Column("taskName", String, nullable=False)
    partName = Column("partName", String, nullable=True, default="---")
    quantity = Column(Integer, nullable=False, default=1)
    partPrice = Column("partPrice", Numeric(10, 2), nullable=False, default=0)
    laborPrice = Column("laborPrice", Numeric(10, 2), nullable=False, default=0)
    totalPrice = Column("totalPrice", Numeric(10, 2), nullable=False, default=0)
    isCompleted = Column("isCompleted", Boolean, nullable=False, default=False)
    completedAt = Column("completedAt", DateTime, nullable=True)
    createdAt = Column("createdAt", DateTime, server_default=func.now())
    updatedAt = Column("updatedAt", DateTime, server_default=func.now(), onupdate=func.now())

    # Relationships
    ticket = relationship("RepairTicket", back_populates="items")
